Examples/CMakeLists.txt

```
@@ -0,0 +1,235 @@
+        return()
+    endif()
+
+    # do not run no-EB tests w/ EB build
+    if(NOT eb AND WarpX_EB)
+        return()
+    endif()
+
+    # cannot run tests with unsupported geometry
+    if(NOT dims IN_LIST WarpX_DIMS)
+        return()
+    endif()
+
+    # cannot run tests with unfulfilled dependencies
+    if(dependency AND NOT TEST ${dependency}.run)
+        return()
+    endif()
+
+    # set dimension suffix
+    warpx_set_suffix_dims(SD ${dims})
+
+    # make a unique run directory
+    file(MAKE_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${name})
+    set(THIS_WORKING_DIR ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${name})
+
+    get_filename_component(INPUTS_ALL ${WarpX_SOURCE_DIR}/${inputs} REALPATH)
+
+    # separate input file (full path) and optional command-line arguments
+    separate_arguments(INPUTS_FILE_ARGS UNIX_COMMAND PROGRAM "${INPUTS_ALL}")
+
+    # get input file (full path)
+    list(GET INPUTS_FILE_ARGS 0 INPUTS_FILE)
+
+    # get optional command-line arguments
+    list(SUBLIST INPUTS_FILE_ARGS 1 -1 INPUTS_ARGS)
+    list(JOIN INPUTS_FILE_ARGS " " INPUTS_FILE_ARGS)
+
+    # get input file's directory
+    get_filename_component(INPUTS_DIR ${INPUTS_FILE} DIRECTORY)
+
+    # copy base input files to run directory
+    file(GLOB INPUTS_BASE "${INPUTS_DIR}/inputs_base_${SD}*")
+    file(COPY ${INPUTS_BASE} DESTINATION ${THIS_WORKING_DIR})
+
+    # copy test input file to run directory
+    file(COPY ${INPUTS_FILE} DESTINATION ${THIS_WORKING_DIR})
+
+    # for restart tests, copy input file of original test (without "_restart" suffix)
+    string(REPLACE "_restart" "" INPUTS_FILE_NORESTART ${INPUTS_FILE})
+    file(COPY ${INPUTS_FILE_NORESTART} DESTINATION ${THIS_WORKING_DIR})
+
+    # Python test?
+    set(python OFF)
+    if(${INPUTS_FILE} MATCHES ".*\.py$")
+        set(python ON)
+    endif()
+
+    # cannot run Python tests w/o Python support
+    if(python AND NOT WarpX_PYTHON)
+        return()
+    endif()
+
+    # set MPI executable
+    set(THIS_MPI_TEST_EXE
+        ${MPIEXEC_EXECUTABLE}
+        ${MPI_ALLOW_ROOT}
+        ${MPIEXEC_NUMPROC_FLAG} ${nprocs}
+        ${MPIEXEC_POSTFLAGS}
+        ${MPIEXEC_PREFLAGS}
+    )
+
+    # set Python executable
+    if(python)
+        set(THIS_Python_EXE ${Python_EXECUTABLE})
+    else()
+        set(THIS_Python_EXE "")
+    endif()
+
+    # test run
+    if(python)
+        # for argparse, do not pass command-line arguments as one quoted string
+        separate_arguments(INPUTS_ARGS UNIX_COMMAND "${INPUTS_ARGS}")
+        add_test(
+            NAME ${name}.run
+            COMMAND
+                ${THIS_MPI_TEST_EXE}
+                ${THIS_Python_EXE}
+                ${INPUTS_FILE}
+                ${INPUTS_ARGS}
+            WORKING_DIRECTORY ${THIS_WORKING_DIR}
+        )
+        # FIXME Use helper function to handle Windows exceptions
+        set_property(TEST ${name}.run APPEND PROPERTY ENVIRONMENT "PYTHONPATH=${PYTHONPATH}:${CMAKE_PYTHON_OUTPUT_DIRECTORY}")
```

The first `${PYTHONPATH}` will be undefined unless we read it first. Do you mean
```suggestion
        set_property(TEST ${name}.run APPEND PROPERTY ENVIRONMENT "PYTHONPATH=$ENV{PYTHONPATH}:${CMAKE_PYTHON_OUTPUT_DIRECTORY}")
```

---

Thanks, corrected in [ffd4d03](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/ffd4d031c42002c30008115d96921116ae3b3da7).